internal/docker/container.go

```diff
@@ -379,6 +387,18 @@ func (r *ContainerRunner) collectResults(artifactCfg config.ArtifactDownload, re
 		rep.Render()
 	}
 
+	slackNtfr.Passed = passed
+
+	if slackNtfr.ShouldSendNotification(notifierCfg) {
+		token, err := r.SlackService.GetSlackToken(context.Background())
```

Comment:
I don't quite understand the design behind splitting the `notifier` and the `slack service`. Two sides of the same coin, imho.

My suggestion for the use of the reporter interface, was such that there is a slack reporter and it is capable of rendering the test results out. The reporter should already be passed in pre-configured, either with the slack token in hand, or the ability to fetch one on the fly when `Render()` is being called.

You really only need one single interface in this PR, which is the `report.Reporter` and slack is one implementation of it.

Reply:
I refactored a bit the code and created a new interface `notification.Reporter` that extends a bit `report.Reporter`.